src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
+                        {
+                            //Handle Failed Operation - Delete                            
+                            PatchStatus status = PatchHandler.TryGet(keyValues, out original, out geterrorMessage);
+                            if (status == PatchStatus.Success)
+                            {
+                                IEdmChangedObject changedObject = HandleFailedOperation(deletedObj, operation, original, keys, errorMessage);
+                                changedObjectCollection.Add(changedObject);
+                                continue;
+                            }
+                        }
+
+                        changedObjectCollection.Add(deletedObj);
+                    }
+                    else
+                    {
+                        EdmEntityObject deltaEntityObject = changedObj as EdmEntityObject;
+
+                        PatchStatus status = PatchHandler.TryGet(keyValues, out original, out geterrorMessage);
+
+                        if (status == PatchStatus.NotFound)
+                        {
+                            operation = DataModificationOperationKind.Insert;
+
+                            if (PatchHandler.TryCreate(changedObj, out original, out errorMessage) != PatchStatus.Success)
+                            {
+                                //Handle failed Opreataion - create
+                                IEdmChangedObject changedObject = HandleFailedOperation(deltaEntityObject, operation, original, keys, errorMessage);
+                                changedObjectCollection.Add(changedObject);
+                                continue;
+                            }
+                        }
+                        else if (status == PatchStatus.Success)
+                        {
+                            operation = DataModificationOperationKind.Update;
+                        }
+                        else
+                        {
+                            //Handle failed operation 
+                            IEdmChangedObject changedObject = HandleFailedOperation(deltaEntityObject, operation, null, keys, geterrorMessage);
+                            changedObjectCollection.Add(changedObject);
+                            continue;
+                        }                                              
+
+                        //Patch for addition/update. 
+                        PatchItem(deltaEntityObject, original);
+
+                        changedObjectCollection.Add(changedObj);
+                    }
+                }
+                catch(Exception ex)
+                {
+                    //Handle Failed Operation
+                    IEdmChangedObject changedObject = HandleFailedOperation(changedObj as EdmEntityObject, operation, originalObj, keys, ex.Message);
+                    
+                    Contract.Assert(changedObject != null);
```

Comment:
why assert here and not other places?

Reply:
in other places we can catch the exception (and less chance as changed object we are getting from request ), here we care creating it and since its in catch block, so just to ensure